src/Router/__test__/buildClientApp.test.tsx

```diff
@@ -19,26 +22,101 @@ describe("buildClientApp", () => {
       ],
     })
 
-    return mount(<ClientApp />)
-  }
-
-  it("resolves with a <ClientApp /> component", async () => {
-    expect((await getWrapper()).html()).toContain("<div>Hello Router</div>")
+    const wrapper = mount(<ClientApp />)
+    expect(wrapper.html()).toContain("<div>Hello Router</div>")
   })
 
   it("accepts an initial route", async () => {
-    expect((await getWrapper("/cv")).html()).toContain("<div>CV Page</div>")
+    const { ClientApp } = await buildClientApp({
+      history: {
+        protocol: "memory",
+      },
+      initialRoute: "/cv",
+      routes: [
+        {
+          path: "/",
+          Component: () => <div>Hello Router</div>,
+        },
+        {
+          path: "/cv",
+          Component: () => <div>CV Page</div>,
+        },
+      ],
+    })
+
+    const wrapper = mount(<ClientApp />)
+    expect(wrapper.html()).toContain("<div>CV Page</div>")
   })
 
   it("bootstraps data from __RELAY_BOOTSTRAP__", async () => {
     window.__RELAY_BOOTSTRAP__ = JSON.stringify([
       ["cacheKey", "found window cache"],
     ])
 
+    const { ClientApp } = await buildClientApp({
+      history: {
+        protocol: "memory",
+      },
+      routes: [
+        {
+          path: "/",
+          Component: () => <div>Hello Router</div>,
+        },
+        {
+          path: "/cv",
+          Component: () => <div>CV Page</div>,
+        },
+      ],
+    })
+
+    const wrapper = mount(<ClientApp />)
     expect(
-      ((await getWrapper())
+      (wrapper
         .find("Boot")
         .props() as any).system.relayEnvironment.relaySSRMiddleware.cache.values()
     ).toContain("found window cache")
   })
+
+  it("passes along initial context values", async done => {
+    const HomeApp = () => {
+      return (
+        <Subscribe to={[AppState]}>
+          {({ state: { system } }) => {
+            expect(Object.keys(system)).toEqual([
+              "history",
+              "routes",
+              "context",
+              "relayEnvironment",
+              "resolver",
+              "currentUser",
+            ])
+            expect(Object.keys(system.context)).toEqual(["mediator", "user"])
+            done()
```

Comment:
Like, 

```
it("injects default renderProps", done => {
    render(
      <Artsy.ContextProvider>
        <Artsy.ContextConsumer>
          {props => {
            setImmediate(() => {
              expect(Object.keys(props).sort()).toEqual([
                "_isNestedInProvider",
                "currentUser",
                "relayEnvironment",
              ])
              done()
            })
            return <div />
          }}
        </Artsy.ContextConsumer>
      </Artsy.ContextProvider>
    )
  })
```
?

Reply:
I was thinking `setImmediate(done)`, but this probably works too.

My reasoning is this; you were telling the test framework that you were done,  when in fact you were _still_ inside the current frame (the function scope) and caller sites may still decide to do work when control returns to them. I think following the semantic reasoning that you call done _outside_ the ongoing call execution isn’t only semantically more correct, it may by chance even save you from annoying bugs. (Annoying in that they are hard to debug as they leak into other concurrent code and avoiding them altogether is always a win.)

Reply:
This makes sense, will update 👍Better to be safe.